### DISCUSSION TITLE
Deeply nested filter fails to work.

### DISCUSSION THREAD

**Discussion by dhurbhat** [PostgREST/postgrest#3031] [Q&A] [answered]
Good day!

I'm struggling with a deeply nested query:
* UserPlaces have Groups; user place is also created by a User profile
* Groups have Members
* Members have a Profile

I want to filter by an `attribute` of Profile for Members in a Group that are at a UserPlace.

The following query **WITHOUT** the filter, does yield the correct JSON output:
`UserPlaces?select=*,Groups(*,GroupMembers(*,UserProfiles(*))),creator:UserProfiles(*)`

BUT attaching it with `&Groups.GroupMembers.UserProfiles.attribute=eq.12345` yields a JSON with the ['Groups']['GroupMembers']['UserProfiles'] value as `null`.

What could I be doing wrong?

**Comment by laurenceisla** [comment]
Hi! Is the `UserProfiles` item with `attribute=eq.12345` included in the response without filters? If it's not, then the filter is working correctly, it will show `null` when it's not found because the top level resource will not be filtered by default.

If you want to show **only**  the results that have that filter, you'll need to use [Top Level Filtering](https://postgrest.org/en/stable/references/api/resource_embedding.html#top-level-filtering) or [Null Filtering](https://postgrest.org/en/stable/references/api/resource_embedding.html#null-filtering-on-embedded-resources). For instance you can change your query to: `UserPlaces?select=*,Groups!inner(*,GroupMembers!inner(*,UserProfiles!inner(*))),creator:UserProfiles(*)`.

If that's not the issue then please add an example with the expected response.

**Reply by dhurbhat** [reply, marked as answer]
Thank you!
I wasn't including the "attribute" in the query so couldn't filter by it.
